### DISCUSSION TITLE
how to create a Fidelity Points program in medusajs ?

### DISCUSSION THREAD

**Discussion by mohamed-achich** [medusajs/medusa#3179] [Q&A]
hello, Is it possible to create a Fidelity Points  program in Medusajs , and how to do it ? :
- do I create a new service
- override the current discount services

thank you,

**Comment by srindom** [comment]
Can you explain further how the points program should work? My guess would be:

1. Customers receive X points per dollar they spend
2. How should products be priced? 
   a. 1 dollar of product price = X points
   b. Products have a point price 
4. How should points be spent?
   a. Customers apply points like a type of credit and pay for any remaining amount with currency
   b. Customers must make purchases fully in either points or currency

I can try to give a high-level outline of how to achieve this if you give me a few more details :)

**Comment by Shlok-Zanwar** [comment]
Hi @srindom 
Can you please help give a high-level outline idea and some starters with the implementation.
Thanks !

**Comment by himanshu-sharma073** [comment]
Hi @srindom ,

By any chance is this available now?

Thanks

**Comment by arceliver** [comment]
I'm also looking for a store credit system made inside medusa, where the customers can have this credits added to their balance and use this to pay for products, but products can have a different value between real currency and store credits. Is this possible to do or to develop?
